src/Lecture4.hs

```diff
@@ -226,7 +289,10 @@ implement the next task.
 -}
 
 combineRows :: NonEmpty Row -> Stats
-combineRows = error "TODO"
+combineRows rows = concat1 $ fmap rowToStats rows
+  where
+    concat1:: (Semigroup a) => NonEmpty a -> a
+    concat1 = foldr1 (\x !acc -> x <> acc)
```

Comment:
Alternatively, you can use `foldl'` here to avoid handling accumulator strictness manually:

```suggestion
    concat1 (x :| xs) = foldl' (<>) x xs
```